src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/common/actions/haflow/BaseHaFlowPathRemovalAction.java

```diff
@@ -56,7 +56,7 @@ protected void removeFlowPaths(HaPathIdsPair haPathIdsPair,
                 if (oldSubPath != null) {
                     log.debug("Removed HA-flow sub path {}", oldSubPath);
                     updateIslsForFlowPath(oldSubPath);
-                    HaFlowHistoryService.using(carrier).save(HaFlowHistory.withTaskId(correlationId)
+                    FlowHistoryService.using(carrier).save(HaFlowHistory.of(correlationId)
```

Comment:
May we not create a new FlowHistoryService object for every iteration here?

Reply:
Small Java objects are cheap, there are no long-running or heavy jobs done in the constructor, so it is not a big deal. There is a bigger task of unifying the carrier, once it is done, there will be no need for the using() static constructor, and it'll be easier to refactor further. 
But it is a good point! I would rather delegate it DI container so that app logic doesn't care about the scopes and lifecycle  of the service class.

Reply:
I created a local variable for the flow history service.